src/main/java/com/greedobank/cards/service/BankAccountService.java

```diff
@@ -0,0 +1,145 @@
+package com.greedobank.cards.service;
+
+import com.greedobank.cards.dao.BankAccountDAO;
+import com.greedobank.cards.dto.BankAccountCreationDTO;
+import com.greedobank.cards.dto.BankAccountDTO;
+import com.greedobank.cards.dto.BankAccountGetBalanceDTO;
+import com.greedobank.cards.dto.BankAccountReplenishDTO;
+import com.greedobank.cards.dto.BankAccountWithdrawDTO;
+import com.greedobank.cards.dto.BankAccountWithdrawOnlineDTO;
+import com.greedobank.cards.exception.InsufficientFundsException;
+import com.greedobank.cards.exception.InvalidInputException;
+import com.greedobank.cards.exception.NotActiveException;
+import com.greedobank.cards.exception.NotFoundException;
+import com.greedobank.cards.mapper.BankAccountMapper;
+import com.greedobank.cards.model.BankAccount;
+import com.greedobank.cards.model.Card;
+import com.greedobank.cards.utils.ResponseMessages;
+import lombok.RequiredArgsConstructor;
+import org.iban4j.CountryCode;
+import org.iban4j.Iban;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+
+@Service
+@RequiredArgsConstructor
+public class
+BankAccountService {
+    private final BankAccountDAO bankAccountDAO;
+    private final BankAccountMapper bankAccountMapper;
+    private final ValidationCardService validationCardService;
+
+    public BankAccountDTO create(BankAccountCreationDTO bankAccountCreationDTO) {
```

Comment:
why does service operate with DTOs? MVC and MVP may seem not that clear in this scope, but what I suggest is to read more about this and e.g. delegate it to controller layer
I suggest to check out some of the resources referenced in [this thread](https://stackoverflow.com/questions/21554977/should-services-always-return-dtos-or-can-they-also-return-domain-models) and I will quote this statement from the comments:

> what if your system interfaces with several other systems or client types, each requiring their own DTO? How do you know which DTO a method of your application service should return?

Reply:
Done